File: 7kyu/(7 kyu) String ends with/(7 kyu) String ends with.java
// 1 Plain solution
public class Kata {
  public static boolean solution(String str, String ending) {
    return str.length() >= ending.length()
      ? str.substring(str.length() - ending.length()).equals(ending)
      : false;
  }
}

// 2 Clever solution
public class Kata {
  public static boolean solution(String str, String ending) {
    return str.endsWith(ending);
  }
}

File: 7kyu/(7 kyu) String ends with/(7 kyu) String ends with.js
// #1
// function solution(str, ending) {
//   const result = str.substr(str.length - ending.length, str.length);
//   return ending === result;
// }
// #2
// function solution(str, ending) {
//   return ending === str.substr(-ending.length);
// }
// #3
const solution = (str, ending) => str.endsWith(ending);
